email-ballerina/imap_listener_endpoint.bal

```diff
@@ -144,34 +122,51 @@ public class ImapListener {
     public isolated function register(service object {} emailService, string? name) {
         register(self, emailService);
     }
+
+    # Close the IMAP server connection.
+    # ```ballerina
+    # email:Error? closeResult = emailClient->close();
+    # ```
+    #
+    # + return - A `email:Error` if it can't close the connection or else `()`
+    isolated function close() returns Error? {
+        error? stopResult = self.stop();
+        return externListenerClose(self);
+    }
 }
 
-final service isolated object{} imapAppointmentService = service object {
-    remote isolated function onTrigger(ImapListener l) {
-        var result = l.poll();
+class Job {
+
+    *task:Job;
+    private ImapListener imapListener;
+
+    public function execute() {
+        var result = self.imapListener.poll();
         if (result is error) {
-            log:printError("Error while executing poll function", err = result);
+            log:printError("Error while executing poll function", 'error = result);
         }
     }
-};
+
+    public isolated function init(ImapListener imapListener) {
+        self.imapListener = imapListener;
+    }
+}
 
 # Configuration for Email listener endpoint.
 #
 # + host - Email server host
 # + username - Email server access username
 # + password - Email server access password
-# + pollingIntervalInMillis - Periodic time interval to check new update
+# + pollingInterval - Periodic time interval (in seconds) to check new update
 # + port - Port number of the IMAP server
 # + security - Type of security channel
-# + cronExpression - Cron expression to check new update
 # + secureSocket - Secure socket configuration
 public type ImapListenerConfig record {|
     string host;
     string username;
     string password;
-    int pollingIntervalInMillis = 60000;
+    decimal pollingInterval = 60;
     int port = 993;
     Security security = SSL;
-    string? cronExpression = ();
-    SecureSocket secureSocket?;
+    ClientSecureSocket secureSocket?;
```

Comment:
If the same secure socket config is used for both listener and client shall we keep the record name as `SecureSocket`. We don't need to update in that case..

Reply:
Reverted back to `SecureSocket`.